packages/forma-36-react-components/src/components/Tooltip/Tooltip.tsx

```diff
@@ -164,19 +172,23 @@ export const Tooltip = ({
         ref={elementRef}
         className={targetWrapperClassName}
         onMouseEnter={(evt: MouseEvent) => {
-          setShow(true);
+          if (!insidePopover) setShow(true);
           if (onMouseOver) onMouseOver(evt);
         }}
         onMouseLeave={(evt: MouseEvent) => {
-          setShow(false);
+          setTimeout(() => { 
+            if (!insidePopover) setShow(false);
+          }, delay)
           if (onMouseLeave) onMouseLeave(evt);
```

Comment:
I'm not sure if `onMouseLeave()` , `onMouseOver()`, etc,  should be called inside the `setTimeout()`, after the delay? 

Also whether it should be called when the tooltip stays open (as the cursor is now on the content) but the cursor has already left the container.

Reply:
Yep I'm not sure as well, but looking into the code makes me think if we should do something with an effect

What I mean is something like we need to track if the mouse is over the target and/or the tooltip. And then depending on the prop you created we hide the tooltip when the mouse is not over any of them

```jsx
const [isHoveringTarget, setIsHoveringTarget] = useState(false)
const [isHoveringContent, setIsHoveringContent] = useState(false)

useEffect(() => {
  if (closeOnMouseLeave) {
    // change show value considering only target hover
    setShow(isHoveringTarget)
  } else {
    // change show value considering hover on both elements
    setShow(isHoveringContent || isHoveringTarget)
  }
}, [closeOnMouseLeave, isHoveringTarget, isHoveringContent])


<Container
  onMouseEnter={() => setIsHoveringTarget(true)}
  onMouseLeave={() => setIsHoveringTarget(false)}
/>
{show && (
  <span
    onMouseEnter={() => setIsHoveringContent(true)}
    onMouseLeave={() => setIsHoveringContent(false)}
  >
    {content}
  </span>
)}

```

I did not test this btw, but I think something like this might work

You are already doing something similar by using that `insidePopover` variable
But I think if you make the logic as a side effect of changing the state then you won't need the timeouts

wdyt?

Reply:
@gui-santos tbh, I'm not sure if we need all that magic 🧙🏻‍♀️I'd just keep it visible on hover at all times, do we really need the switch prop? 
so what I mean is move the content inside the container, and then everything is controlled by one `onMouseEnter` and one `onMouseLeave` ✨wdyt?

Reply:
We can do that, but then we won't be able to use React Portals with the Tooltip in the future which means the tooltip will always calculate its position within its parent

Also I don't think there is any magic, it's just react hahahaha 
you have a state, when it changes it re-renders. If you want to do something related with the new state, it's a side effect
https://reactjs.org/docs/hooks-effect.html

I was just suggesting something to remove the necessity of the timeouts inside the mouse events
And using the react state system seemed pretty fair, it's already what the variable `insidePopover` does with the disadvantages of being just a variable and not a state

Reply:
ok cool, why do we need the portal though? I think for accessibility and keyboard focus order, It's always better to avoid using portal and keep everything encapsulated in one container

Reply:
The dropdown is using portals now for example

Without it, this is what popper.js does
![Screenshot 2020-11-13 at 16 30 43](https://user-images.githubusercontent.com/6597467/99089427-98fa5c80-25cd-11eb-9960-79089312bcee.png)

You see, in relation to the window, we have enough space for the tooltip to show at the bottom side of its target. But popper checks the target's parent and it sees that there is no room for the tooltip so it shows the tooltip over the left side of the target

Reply:
@gui-santos I will try the suggested way using the effects and see if I can get rid of the setTimteout() 👍

The reason I added `setTImeout()` is that there is a gap between the Container and the Tooltip content, and as soon as the cursor leaves the Container and before reaching the Content, the `mouseLeave()` on Container is triggered. 
This then hides the Content immediately, hence the `mouseEnter()` on the Content (`<span >...`) never gets triggered. I wanted to have some delay before the tooltip gets hidden, so that we could actually detect the `mouseEnter()` over the Content.

Reply:
Ahhhh yes you told me this before! I totally forgot about this. Sorry!
Yes I guess the timeout needs to exist